### PR TITLE
Context should inform about adding new editors to it

### LINKED ISSUE #6089: Context should inform about adding new editors to it
## 📝 Provide a description of the new feature

`Context` stores its editors under a private property names `_editors`. At the moment there's no way to react when a new editor is dynamically added to the context. This could be useful for some features and integrations.

My proposals are:

1.  Make `editors` a public property and make it a `Collection` instance.
2.  Fire an event whenever an editor instance is added to the `Context`.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: `Context#editors` is now public property and a `Collection` instance which allows for listening to adding an editor to a `Context`. Closes https://github.com/ckeditor/ckeditor5/issues/6089.

---

### Additional information

As mentioned in https://github.com/ckeditor/ckeditor5/issues/6089 there were two ways to solve this issue. I've chosen to switch `editors` to `Collection` instance. Not sure if it was the right call, though.

Basically, there are multiple places in our tests where we destroy the editor multiple times. This caused crashes (because `Collection` throws when you try to remove a non-existing item). I gave up and simply put the removing in an `if`. But I don't like it.

So feel free to criticize this and ask me to implement it the other way around (adding a custom `event:add` to `Context`).

### CODE REVIEW

File: tests/editor/utils/attachtoform.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ describe( 'attachToForm()', () => {
 		submitStub.restore();
 		form.remove();
 
-		return editor.destroy();
+		if ( editor ) {
```
Comment (oleq) [inline tests/editor/utils/attachtoform.js:46]: I'm sort of confused here. First, [you make sure it's OK to destroy the editor twice](https://github.com/ckeditor/ckeditor5-core/pull/210/files#diff-fbd6c571314d640867976e2efb7b4f23R201), then you're fixing tests (seeming unrelated to the PR) to make sure the editor is not destroyed twice. Am I missing something?

BTW: I'm pretty sure there are billions of tests destroying the editor twice across the codebase already.

File: tests/editor/utils/securesourceelement.js [hunk 2]
```diff
@@ -53,6 +53,8 @@ describe( 'secureSourceElement()', () => {
 
 		return editor.destroy()
 			.then( () => {
+				editor = null;
```
Comment (oleq) [inline tests/editor/utils/securesourceelement.js:56]: The same concern I already expressed in `tests/editor/utils/attachtoform.js`.

**Review by oleq — COMMENTED** [review]
TBH I don't see anything wrong with exposing `Context#editors` as a public API. We could [delegate](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_emittermixin-EmitterMixin.html#function-delegate) (hoist) collection events to the `Context` instance but I'm not sure it's worth the hassle.

What I'm concerned about though is some changes in tests I don't get (see my comments).

File: tests/editor/utils/attachtoform.js (hunk 1 above)
Comment (scofalik) [reply tests/editor/utils/attachtoform.js:46]: This is a relict of changes to the code that happened in two steps.

First I did that change in the test here. Then I discovered that I would have to change dozens of tests in multiple repos. So, I gave up and instead of fixing all the places where editor is destroyed twice, I made sure it doesn't crash.

So, this change is not needed. I can revert it, if you insist.

Comment (oleq) [reply tests/editor/utils/attachtoform.js:46]: > So, this change is not needed. I can revert it, if you insist.

Yes, please. Let's keep this PR clean.
